File: todoapp/lib/domain/repository/authentication.dart
import 'package:todoapp/domain/model/user_auth.dart' as $model;

abstract class AuthenticationRepositoryImplement {
  Future<$model.Session> signIn(String email, String password);
  Future<SignUpOutput> signUp(String email, String password);
  Future<void> signOut($model.Session session);
}

class SignUpOutput {
  $model.Session? session;
  $model.UserSignUpStatus status;
  SignUpOutput({required this.session, required this.status});
}

File: todo-web/src/domain/repository/authentication.ts
import { Session, UserSignUpStatus } from "../model/authentication";

export interface AuthenticationRepository {
    signUp(name: string, email: string, password: string): Promise<SignUpOutput>;
    signIn(email: string, password: string): Promise<SignInOutput>;
    signOut(session: Session): Promise<void>;
};

export type SignUpOutput = {
    session?: Session;
    status: UserSignUpStatus;
};

export type SignInOutput = {
    session?: Session;
};

